packages/wrangler/src/dev/use-esbuild.ts

```diff
@@ -84,7 +84,9 @@ export function useEsbuild({
       // on build errors anyway so this is a no-op error handler
     });
 
-    return stopWatching;
```

Comment:
React ignores subsequent setups of the cleanup callback, so this needs to be a function on first run. I'm surprised there was no warning thrown here, I kinda remember us working on it back then. Might've been lost at some point.